packages/api/test/unit/platforms/vtex/clients/commerce.test.ts

```diff
@@ -0,0 +1,109 @@
+  environment: 'vtexcommercestable',
+  channel: '{"salesChannel":"1"}',
+  locale: 'en-US',
+  subDomainPrefix: ['www'],
+  hideUnavailableItems: false,
+  incrementAddress: false,
+  flags: {
+    enableOrderFormSync: true,
+  },
+} as Options
+
+const contextFactory = getContextFactory(apiOptions)
+const context = contextFactory({})
+const getClients = clients.getClients
+
+const fetchAPIMocked = jest.fn()
+const pickupPointsMocked = jest.fn()
+
+jest.mock('../../../../../src/platforms/vtex/clients/fetch.ts', () => ({
+  fetchAPI: async (
+    info: RequestInfo,
+    init?: RequestInit,
+    options?: { storeCookies?: (headers: Headers) => void }
+  ) => fetchAPIMocked(info, init, options),
+}))
+
+beforeEach(() => {
+  fetchAPIMocked.mockClear()
+})
+
+describe('VTEX Commerce', () => {
+  describe('Checkout', () => {
+    describe('Pickup points', () => {
+      it('should succeed with valid geo coordinates', async () => {
+        const validResponse = { paging: {}, items: [] }
+        const geoCoordinates = {
+          latitude: 123,
+          longitude: 456,
+        }
+
+        fetchAPIMocked.mockResolvedValueOnce(validResponse)
+
+        const { commerce } = clients.getClients(apiOptions, context)
+        const result = await commerce.checkout.pickupPoints({
+          geoCoordinates,
+        })
+
+        expect(fetchAPIMocked).toHaveBeenCalledTimes(1)
+        expect(result).toEqual(validResponse)
+      })
+
+      it('should succeed with valid postal code and country', async () => {
+        const country = 'BRA'
+        const postalCode = '123456'
+        const validResponse = { paging: {}, items: [] }
+
+        fetchAPIMocked.mockResolvedValueOnce(validResponse)
+
+        const { commerce } = clients.getClients(apiOptions, context)
+        const result = await commerce.checkout.pickupPoints({
+          country,
+          postalCode,
+        })
+
+        expect(fetchAPIMocked).toHaveBeenCalledTimes(1)
+        expect(result).toEqual(validResponse)
+      })
+
+      it('should throw an error when no params', async () => {
+        jest
+          .spyOn(clients, 'getClients')
+          .mockImplementation((options, context): Clients => {
+            const otherClients = getClients(options, context)
+
+            return {
+              ...otherClients,
+              commerce: {
+                ...otherClients.commerce,
+                checkout: {
+                  ...otherClients.commerce.checkout,
+                  pickupPoints: pickupPointsMocked,
+                },
+              },
+            }
+          })
+
+        const errorMessage = 'Missing required parameters'
+        pickupPointsMocked.mockRejectedValueOnce(new Error(errorMessage))
+
+        const { commerce } = clients.getClients(apiOptions, context)
+
+        expect(fetchAPIMocked).not.toHaveBeenCalled()
+        await expect(commerce.checkout.pickupPoints({})).rejects.toThrowError(
+          errorMessage
+        )
```

Comment:
The idea would be to let the commerce.checkout.pickupPoints fail so we check that it fails (as it should) when no params are sent and that the fetchAPI isn't called in this case:
```suggestion
         const errorMessage = 'Missing required parameters'
        const { commerce } = clients.getClients(apiOptions, context)

        expect(() => commerce.checkout.pickupPoints({})).toThrow(errorMessage)
        expect(() =>
          commerce.checkout.pickupPoints({
            geoCoordinates: undefined,
            postalCode: undefined,
          })
        ).toThrow(errorMessage)
        expect(fetchAPIMocked).not.toHaveBeenCalled()
```
In the suggestion I've added 2 tests (empty obj {} and obj with undefined params) but feel free to separate them if you feel it's best